### PR TITLE
Fix - sticky table headers

### PR DESCRIPTION
**Pull request by jinamshah**
What done looks like

- Calendar, list, and comparison view tables have sticky headers

Where the changes need to be made

- Calendar view
- List view
- Comparison view

### CODE REVIEW

**Comment by jinamshah** [comment]
Sticky headers are not supported for elements without fixed heights, and since we're using grids, we couldn't have provided a fixed height without having two scroll bars - one for page and one for the table. This issue would create problems in all tables.
Hence, the solution is as follows:
Store the widths of the table and sidebar on a page and compare it with window size, if greater, then change the width of the header and footer to accommodate this change in window size. This + overflowX: 'initial' removes the table scrolls (both horizontal and vertical).

Also, crop card has been removed because it was taking too much room and instead crop's scientific name is added to the table and modal open/close is added to the crop label header

**Comment by mikahpinegar** [comment]
Your review app has been deployed and is available at 'http://135.237.20.24:80'.<br><br>[Learn More](https://aka.ms/testwithreviewapps) about how to test and provide feedback for the app.

**Comment by mikahpinegar** [comment]
Your review app has been deployed and is available at 'http://135.237.20.24:80'.<br><br>[Learn More](https://aka.ms/testwithreviewapps) about how to test and provide feedback for the app.

File: src/pages/CropSidebar/CropSidebar.js [hunk 1]
```diff
@@ -295,6 +295,14 @@ const CropSidebar = ({
     filtersList();
   }, [sidebarFilters]);
 
+  // const sidebarRef = useRef(null);
```
Comment (YazidZidane) [inline src/pages/CropSidebar/CropSidebar.js:298]: remove comments if not needed

File: src/pages/MyCoverCropList/MyCoverCropComparison/MyCoverCropComparisonTable.js [hunk 2]
```diff
@@ -107,22 +111,33 @@ const MyCoverCropComparisonTable = () => {
...
+          color: 'primary.main',
+          cursor: 'pointer',
+          textDecoration: 'none',
+          '&:hover': {
+            textDecoration: 'underline',
+            color: 'primary.dark',
+          },
+          transition: 'color 0.3s ease',
+        }}
+        onClick={() => handleModalOpen(crop)}
```
Comment (YazidZidane) [inline src/pages/MyCoverCropList/MyCoverCropComparison/MyCoverCropComparisonTable.js:131]: I found that this onClick function is making a lots of same api calls which will freeze the page for a few seconds. Can you look into this and try to solve that?

File: src/pages/CropSelector/CropCalendarView/CropCalendarView.js [hunk 3]
```diff
@@ -133,21 +134,34 @@ const CropCalendarView = ({
...
   }, [cropDataRedux]);
 
+  const tableRef = useRef(null);
+  const dispatchRedux = useDispatch();
+  useEffect(() => {
+    if (tableRef.current) {
+      // let tableWidth = tableRef.current.offsetWidth;
+      const tableWidth = tableRef.current.scrollWidth;
+      // tableWidth += 1000;
+      // console.log('table width in page', tableWidth, typeof (tableWidth));
```
Comment (YazidZidane) [inline src/pages/CropSelector/CropCalendarView/CropCalendarView.js:144]: remove comments if not needed

File: src/pages/CropSelector/CropTable/CropTable.js [hunk 4]
```diff
@@ -112,10 +113,21 @@ const CropTable = ({
...
   }, []);
 
+  const tableRef = useRef(null);
+  const dispatchRedux = useDispatch();
+  useEffect(() => {
+    if (tableRef.current) {
+      // const tableWidth = tableRef.current.offsetWidth;
+      const tableWidth = tableRef.current.scrollWidth;
+      // tableWidth += 1000;
+      // console.log('table width in page', tableWidth, typeof (tableWidth));
```
Comment (YazidZidane) [inline src/pages/CropSelector/CropTable/CropTable.js:123]: remove comments if not needed

File: src/pages/MyCoverCropList/MyCoverCropComparison/MyCoverCropComparisonTable.js (hunk 2 above)
Comment (YazidZidane) [reply src/pages/MyCoverCropList/MyCoverCropComparison/MyCoverCropComparisonTable.js:131]: Just noticed that in this same function the `<CropDetailsModal>` is created for each crop so there'll be a lot of same modals created when `onCLick` triggers. Can you move the modal outside of this function, so there'll be only one modal popup when clicked?

Comment (jinamshah) [reply src/pages/MyCoverCropList/MyCoverCropComparison/MyCoverCropComparisonTable.js:130]: I've moved cropdetailsmodal outside but kept the onclick function. It'll just send the data that needs to be attached to the modal

**Comment by mikahpinegar** [comment]
Your review app has been deployed and is available at 'http://135.237.20.24:80'.<br><br>[Learn More](https://aka.ms/testwithreviewapps) about how to test and provide feedback for the app.

**Comment by mikahpinegar** [comment]
Your review app has been deployed and is available at 'http://135.237.20.24:80'.<br><br>[Learn More](https://aka.ms/testwithreviewapps) about how to test and provide feedback for the app.

**Review by YazidZidane — APPROVED** [review]

**Pull request merged**